components/gitpod-protocol/src/messaging/proxy-factory.ts

```diff
@@ -217,11 +223,13 @@ export class JsonRpcProxyFactory<T extends object> implements ProxyHandler<T> {
                     new Promise((resolve, reject) => {
                         try {
                             if (isNotify) {
+                                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
                                 connection.sendNotification(p.toString(), ...args);
                                 resolve(undefined);
                             } else {
+                                // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
                                 const resultPromise = connection.sendRequest(p.toString(), ...args) as Promise<any>;
-                                resultPromise.catch((err: any) => reject(err)).then((result: any) => resolve(result));
+                                resultPromise.then((result: any) => resolve(result)).catch((err: any) => reject(err));
```

Comment:
I believe this was the wrong order. cc @akosyakov @geropl

Reply:
I think you can also do `resultPromise.then(resolve, reject)`